documentation/docs/explanations/concurrency.md

```diff
@@ -0,0 +1,80 @@
+
+The idea is quite straightforward. A global clock is used and incremented every time a transaction terminates. 
+Shared memory/variables are tagged with a local clock value corresponding to the moment they were last modified by a transaction commitment.
+Transactions are computed on copies of these shared memory/variables.
+At the end of each transaction the system needs to check that all the shared memory/variables used during the transaction have not been tampered by another transaction to keep a consistent behaviour. 
+To check such thing the local clock of the copies used for the transaction are checked against the local clock of the original shared memory/variables.
+If the check detect any inconsistency between clock values then the transaction is reset and computed again.
+If the check is successful the copies are committed and replace the original shared memory/variables and they are also given a new local clock value by the global clock.
+
+This approach advantage is that the idea is relatively simple and implement the STM model. The downside is that the implementation we developed of TL2 is painfully slow and usually going single thread is faster than using multiple threads. 
+We suspect that this poor performance are due to multiple interleaving of threads which force the transactions to be reset multiple times. Further investigation is required to confirm this idea.
+
+
+## Read-Log-Update (RLU)
+
+RLU is an extension of the more famous _Read-Copy-Update_ (RCU) that has been widely adopted in the Linux kernel.
+
+RLU was first presented in 2015 in the paper [Read-Log-Update: A Lightweight Synchronization Mechanism for Concurrent Programming](http://sigops.org/sosp/sosp15/current/2015-Monterey/printable/077-matveev.pdf). 
+
+Contrary to TL2 previously presented, RLU is a blocking algorithm.
+Threads do their computations locally and synchronize with each other using their local clock (when the computations started) and a common global clock.
+A thread logs the new value of any shared memory that it has modified.
+To read shared memory a thread either does it directly in the memory or go fetch it in another thread logs when it has been locked and is being modified.
+When a thread wants to commit its computations to the memory it has to wait that all the other threads that have started with an old memory state to terminate beforehand.
+The upside of such approach is that a thread never needs to redo its computation since it has waited the correct timing to commit its work.
+In return threads have to spend some time idle since they have to wait. 
+However the whole RLU algorithm is lock-free meaning that at any moment there is always a thread that advances in its computation and the whole system cannot be hard stuck.
```

Comment:
```suggestion
Contrary to [TL2](#tl2), RLU is a blocking algorithm.
Threads do their computations locally and synchronize with each other using their local clock (when the computations started) and a common global clock.
A thread logs the new value of any shared memory it has modified.

To read shared memory, a thread either does it directly in the memory or fetches it in another thread logs when it has been locked and is being modified.
When a thread wants to commit its computations to the memory, it has to wait for all the other threads that have started with an old memory state to terminate.

The upside to this approach is that a thread never needs to redo its computation since it has to wait for the correct timing to commit its work.

In return, threads may be idle for some time as they have to wait.
However, the whole RLU algorithm is lock-free. This means that at any moment there is always a thread that advances in its computation, and the whole system cannot be hard-stuck.
```